tests/unit/test-class-wc-payment-gateway-wcpay.php

```diff
@@ -3969,6 +3969,50 @@ public function is_proper_intent_used_with_order_returns_false() {
 		$this->assertFalse( $this->card_gateway->is_proper_intent_used_with_order( WC_Helper_Order::create_order(), 'wrong_intent_id' ) );
 	}
 
+
+	public function test_get_recommended_payment_method() {
+		$this->mock_wcpay_account
+			->expects( $this->once() )
+			->method( 'get_recommended_payment_methods' )
+			->with( 'US' );
+		$this->card_gateway->get_recommended_payment_methods( 'US' );
+	}
+
+
+	public function get_recommended_payment_method_no_country_code_provider() {
+		return [
+			'provider connected'     => [ true, 'test' ],
```

Comment:
We should also test for the store base country fallback.

Reply:
we test the base country fallback with `provider not connected`

Reply:
> we test the base country fallback with provider not connected
But how do you assert that the country used is the WC base country location?

Reply:
if nothing is set, in `wc_get_base_location` it is set `US` as default country. however, for visibility, I added a specific default country in [9222e44](https://github.com/Automattic/woocommerce-payments/pull/9825/commits/9222e44ebd875b896099237eb064eee1cf5f44b5)

Reply:
Thank you. It is good practice to keep our unit tests as contained as possible.

Also, make sure to remove any added filters otherwise we will have flaky tests. We can use `remove_all_actions` and `remove_all_filters` as a cludge (but we need to be careful not to unhook general arrangements that should remain in place). For this reason, it is best to save the function reference in a variable, hook it at the beginning on the test and unhook it at the end - [here is an example](https://github.com/woocommerce/woocommerce/blob/dbf2886c65c0592f92977c8ef36cb6e1a43c736b/plugins/woocommerce/tests/php/src/Internal/Admin/Settings/PaymentsRestControllerIntegrationTest.php#L202-L202).

Reply:
done in [34d7fa6](https://github.com/Automattic/woocommerce-payments/pull/9825/commits/34d7fa660420b9d6353f44a924b01801da5e7a0f) . thx